sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,501 @@
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        """
+        return self._boundary_conditions
+
+    def apply_boundary_conditions(self, **bcs):
+        """
+        Takes the boundary conditions of the beam bending problem as input.
+        The boundary conditions should be passed as keyworded arguments.
+        It is suggested to use ``moment``, ``slope`` and ``deflection`` as the
+        keywords while providing the boundary conditions of the moment curve,
+        slope curve and the deflection curve respectively as inputs.
+
+        Outputs a dictionary.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0)]
+
+        """
+        if bcs['moment']:
+            self._boundary_conditions['moment'].extend([m_bcs for m_bcs in bcs['moment']])
+        if bcs['slope']:
+            self._boundary_conditions['slope'].extend([s_bcs for s_bcs in bcs['slope']])
+        if bcs['deflection']:
+            self._boundary_conditions['deflection'].extend([d_bcs for d_bcs in bcs['deflection']])
+
+    def apply_moment_boundary_conditions(self, *m_bcs):
+        """
+        Takes only the moment boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_moment_boundary_conditions((4, 3), (5, 0))
+        [(0, 4), (4, 0), (4, 3), (5, 0)]
+
+        """
+        for bcs in m_bcs:
+            self._boundary_conditions['moment'].append(bcs)
+        return self._boundary_conditions['moment']
```

Comment:
If the class is mutable and these are just for storing the boundary conditions internally I don't think it is useful to return them. I don't think anyone will ever even use the boundary conditions in that form, so that's another reason not to return them.